### PR TITLE
fbgemm-xpu: add CI workflow

### PR DESCRIPTION
**Pull request by flezaalv**
cc: @aagalleg, @manuelhsantana, @luis-real

### CODE REVIEW

**Comment by dvrogozh** [comment]
There are couple attempts to break current CI workflow into dedicated torchcodec and fbgemm parts (https://github.com/intel/torchlib-xpu/pull/86 and https://github.com/intel/torchlib-xpu/pull/88). Unfortunately we will have some issues with this approach centered around the fact that we have one more package to take care about - `torchlib-xpu` which is a virtual package depending on other. If you will think thru that you will see that it's not quite trivial to satisfy all the requirements at once such as aligned pytorch version to use across the whole project and having CI in separate workflows for each project.
 
Overall, I suggest a little bit different strategy. I think we should not break current monolithic CI workflow. I believe that it's better to have single pass to build all the components - this part is cheap enough and 90% the same across both projects. What we need to do is to optimize test time. For that I suggest to add skips of the test jobs depending on the modified paths. Here is respective PR, please, take a look:
* See https://github.com/intel/torchlib-xpu/pull/93

@flezaalv ,  can you, please, modify the PR building on top of https://github.com/intel/torchlib-xpu/pull/93?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -328,3 +338,210 @@ jobs:
       uses: pypa/gh-action-pypi-publish@release/v1
       with:
         packages-dir: _publish
+  
+  # ============================================================================
+  # FBGEMM-XPU JOBS
+  # The following jobs build, test, and publish fbgemm-xpu package
+  # ============================================================================
+  build-fbgemm:
```
Comment (dvrogozh) [inline .github/workflows/ci.yml]: you don't need `build-fbgemm` or `compose-fbgemm` because `build` and `compose` already cover fbgemm - these steps build everything.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -328,3 +338,210 @@ jobs:
...
+      run: |
+        # TODO: Apply XPU patches here
+        echo "Patches will be applied here"
+    - name: 'Test fbgemm-xpu'
+      working-directory: fbgemm
+      run: |
+        # TODO: Run pytest commands to run the XPU tests here
+        echo "Instructions to run the tests will be added here"
+
+  publish-fbgemm:
```
Comment (dvrogozh) [inline .github/workflows/ci.yml]: dedicate fbgemm publish step is also not needed.

**Comment by flezaalv** [comment]
> There are couple attempts to break current CI workflow into dedicated torchcodec and fbgemm parts (#86 and #88). Unfortunately we will have some issues with this approach centered around the fact that we have one more package to take care about - `torchlib-xpu` which is a virtual package depending on other. If you will think thru that you will see that it's not quite trivial to satisfy all the requirements at once such as aligned pytorch version to use across the whole project and having CI in separate workflows for each project.
> 
> Overall, I suggest a little bit different strategy. I think we should not break current monolithic CI workflow. I believe that it's better to have single pass to build all the components - this part is cheap enough and 90% the same across both projects. What we need to do is to optimize test time. For that I suggest to add skips of the test jobs depending on the modified paths. Here is respective PR, please, take a look:
> 
> * See [ci: restrict tests to changes in respective directories #93](https://github.com/intel/torchlib-xpu/pull/93)
> 
> @flezaalv , can you, please, modify the PR building on top of #93?

I will do

File: packages/fbgemm-xpu/README.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ python -c "import fbgemm_xpu; print(fbgemm_xpu.__version__)"
 
 Environment variables will be added as new FBGEMM operators are integrated into this project.
 
-## Known limitations
+## Known limitations
```
Comment (dvrogozh) [inline packages/fbgemm-xpu/README.md]: You are adding this change just to trigger the fbgemm ci test step, rigjht? Do it differently: add an empty file `packages/fbgemm-xpu/patches/.gitkeep`. We will use `patches` folder to store patches for upstream fbgemm.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -321,6 +423,7 @@ jobs:
       run: |
         rm -rf _publish && mkdir _publish
         cp ./_wheels/${{ env.WHEELS }}-${{ env.PR_OR_SHA }}-${{ env.UNIQUE_ID }}/torchcodec_xpu-*.whl _publish/
+        cp ./_wheels/${{ env.WHEELS }}-${{ env.PR_OR_SHA }}-${{ env.UNIQUE_ID }}/fbgemm_xpu-*.whl _publish/
```
Comment (dvrogozh) [inline .github/workflows/ci.yml]: Drop this line, please. We will add it once we will be ready to start publishing fbgemm-xpu wheels on PyPI. We are not ready yet.

**Review by dvrogozh — COMMENTED** [review]
You need to rebase the

File: packages/fbgemm-xpu/pyproject.toml [hunk 5]
```diff
@@ -14,6 +14,9 @@ readme = "README.md"
 requires-python = ">=3.10"
 license = { file = "LICENSE" }
 authors = [
+    { name = "Intel Corp." }
```
Comment (dvrogozh) [inline packages/fbgemm-xpu/pyproject.toml]: Why this change? To be honest I would prefer to have specific name for the author rather than Intel. But if you guys don't want names to show up at PyPI, then I can understand that.

**Review by dvrogozh — COMMENTED** [review]
Can you, please, move `packages/fbgemm-xpu/test/patches/.gitkeep‎` to `packages/fbgemm-xpu/patches/.gitkeep‎`? This will correlate with the placement of the patches for torchcodec.

**Comment by flezaalv** [comment]
> Can you, please, move `packages/fbgemm-xpu/test/patches/.gitkeep‎` to `packages/fbgemm-xpu/patches/.gitkeep‎`? This will correlate with the placement of the patches for torchcodec.

Done.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -310,8 +310,100 @@ jobs:
...
+    - name: 'Install pip'
+      run: |
+        conda install pip
+    - name: 'Install fbgemm-xpu'
+      run: |
+        pwd
+        ls -l 
+        ls -l ./_wheels/
+        pwd
+        find ./_wheels/ -name "*.whl"
```
Comment (dvrogozh) [inline .github/workflows/ci.yml]: That's more of a debug. I suggest to drop.
```suggestion
```

**Review by dvrogozh — COMMENTED** [review]
@flezaalv , I think this PR is almost ready. Please, take a look and reply at last couple comments and mark as ready for review.

File: packages/fbgemm-xpu/pyproject.toml (hunk 5 above)
Comment (dvrogozh) [reply packages/fbgemm-xpu/pyproject.toml]: I'll revert that. Let's focus on the CI change and do in other PR if needed.

**Review by dvrogozh — DISMISSED** [review]
LGTM

**Review by luis-real — DISMISSED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -310,8 +310,95 @@ jobs:
...
+        path: _wheels/
+    - name: 'Checkout'
+      uses: actions/checkout@v7
+      with:
+        path: torchlib-xpu
+    - name: 'Checkout'
+      uses: actions/checkout@v7
+      with:
+        repository: pytorch/FBGEMM
+        ref: v1.7.0
```
Comment (aagalleg) [inline .github/workflows/ci.yml]:
```suggestion
        ref: v1.8.0
```

**Review by dvrogozh — APPROVED** [review]

**Review by aagalleg — APPROVED** [review]
LGTM
